app/Meta.py

```diff
@@ -132,40 +130,32 @@ def insert_sites(fileName,shp_file,dbf_file,prj_file,shx_file):
 
         #Reading the shapefile as DataFrame
         data_g1=gpd.read_file(shp_file_target)              
-        
-        #data['geometry'] = data.geometry.astype(str)
-        
-        """for i in range(data_g.shape[0]):
-            p=data_g.iat[i,10]
-            geos.lgeos.GEOSSetSRID(p._geom, 4326)
-            data.iat[i,6]=p.wkb_hex"""
 
         data_g  = data_g1.to_crs({'init': 'epsg:4326'})
 
-        flat_list = []
         for index, row in data_g.iterrows():
+            flat_list = []
             for pt in list(row['geometry'].exterior.coords):
                 pt=pt+(115,)
```

Comment:
This line of code adds another dimension to the loaded point. This causes problems when the points already have a Z value when loaded from the shapefile (point has 3 dimensions); after this line the point will have 4 dimensions and cause a failure

Reply:
Another comment on this is that every point that doesn't have a Z value will be assigned an elevation of 115 meters. IMO it would be better to default to zero or some other value indicating that it's probably not correct (unless you're at sea level)